src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1433,7 +1433,7 @@ tx_gateway_frontend::do_end_txn(
     }
     tx = r.value();
 
-    auto ongoing_tx = co_await stm->mark_tx_ongoing(tx.id);
+    auto ongoing_tx = co_await stm->mark_tx_ongoing(term, tx.id);
```

Comment:
didn't understand this change, this `term` is the current term and we are comparing with `tx.etag` which could be older, you mean term change is not allowed?

Reply:
if leadership bounces between current node and another and back during the execution of mark_tx_ongoing we may accidentally mark a newer tx as ongoing assuming it's the current.